### PR TITLE
Cupertino { TabScafold, TextSelection, TextField } dark mode & minor fidelity update

### PR DESCRIPTION
**Pull request by LongCatIsLooong**
## Description

Updated dark mode colors. Text editing menu color is still not correct: https://github.com/flutter/flutter/issues/41507. The color of the clear button seems to have changed  in iOS 13.1 so might be different from what you see on iOS 13.0 devices/simulators.

## Screenshots

| dark | light | before |
| --- | --- | --- |
| Simulator Screen Shot - iPhone Xʀ - 2019-09-27 at 15 10 15 | Simulator Screen Shot - iPhone Xʀ - 2019-09-27 at 15 08 08 | Simulator Screen Shot - iPhone Xʀ - 2019-10-02 at 17 52 19 |

## Related Issues

https://github.com/flutter/flutter/issues/35541

## Tests

I added the following tests:

- dark mode background color
- updated goldens: https://github.com/flutter/goldens/pull/47

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [x] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/cupertino/text_selection.dart [hunk 1]
```diff
@@ -31,13 +32,16 @@ const double _kToolbarContentDistance = 8.0;
...
+// Values extracted from https://developer.apple.com/design/resources/.
 // The height of the toolbar, including the arrow.
 const double _kToolbarHeight = 43.0;
-const Color _kToolbarBackgroundColor = Color(0xEB202020);
-const Color _kToolbarDividerColor = Color(0xFF808080);
 const Size _kToolbarArrowSize = Size(14.0, 7.0);
-const EdgeInsets _kToolbarButtonPadding = EdgeInsets.symmetric(vertical: 10.0, horizontal: 18.0);
 const Radius _kToolbarBorderRadius = Radius.circular(8);
+// Colors extracted from https://developer.apple.com/design/resources/.
+// TODO(LongCatIsLooong): https://github.com/flutter/flutter/issues/41507.
```
Comment (justinmc) [inline packages/flutter/lib/src/cupertino/text_selection.dart]: Thanks for noticing this and creating the issue.

**Review by justinmc — APPROVED** [review]
LGTM 👍

**Review by HansMuller — APPROVED** [review]
LGTM (too)

**Review by Piinks — APPROVED** [review]
LGTM!

**Pull request merged**